pkg/controller/endpointslice/reconciler.go

```diff
@@ -73,6 +78,14 @@ func (r *reconciler) reconcile(service *corev1.Service, pods []*corev1.Pod, exis
 	for _, existingSlice := range existingSlices {
 		// service no longer supports that address type, add it to deleted slices
 		if _, ok := serviceSupportedAddressesTypes[existingSlice.AddressType]; !ok {
+			if r.topologyCache != nil {
+				svcKey, err := serviceControllerKey(existingSlice)
+				if err != nil {
+					klog.Warningf("Couldn't get key to remove EndpointSlice from topology cache %+v: %v", existingSlice, err)
+				}
+				r.topologyCache.RemoveHints(svcKey, existingSlice.AddressType)
```

Comment:
do we still remove the hints from the cache in case serviceControllerKey() returns an error?

Reply:
Updated, thanks!